### PR TITLE
Update developer documentation, templates, and README

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

- Update the templates to be more concise
    - `ISSUE_TEMPLATE.md`
    - `PULL_REQUEST_TEMPLATES.md`
- Update the `README.md`
    - Remove outdated information
    - Include installation guide and authentication examples
- Update links in `CONTRIBUTING.md`
- Add documentation on patterns and standards
    - Move developer guide to `documentation` folder
    - Add design guidelines to `documentation` folder
- Rename `release-notes` folder to `migration-guides`
- Move testing documentation to `testing-docs` folder

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#making-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: documentation/development-docs/should-process-confirm-impact.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+Users who specify the `-Confirm` parameter on the command prompt will automatically be prompted for **any** call to `ShouldProcess`, and can decide whether to continue processing. Users can increase or decrease their `$ConfirmPreference` level to decide which kinds of changes will be confirmed automaticlaly without user interaction.
+
+Users can specify the `-WhatIf` parameter to see all the `ShouldProcess` prompts that would occur during exeuction of a cmdlet, without actually making any changes.
+
+Some cmdlets required additional confirmation. For example, if a cmdlet would destroy existing resources in some circumstances, the cmdlet might detect that condition and prompt the user to verify before continuing. Overwriting an existing resource during resource creation, overwriting a file when downloading data, deleting a resource that is currently in use, or deleting a container that contains additional resources are all example of this pattern. To implement additional confirmation, and allow scripts to opt out of additional prompts, the above pattern is enhanced with calles to `ShouldContinue()` and the `Force` parameter:
+
+```cs
```
Comment (twitchax) [inline documentation/development-docs/should-process-confirm-impact.md]: Does this need to be `csharp`?

File: documentation/development-docs/should-process-confirm-impact.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    if (Force || ShouldContinue(shouldContinueWarning, shouldContinueCaption))
+    {
+        // make change that required confirmation
+    }
+}
+```
+
+Notice that to _automatically skip prompts_ for such a cmdlet requires the user to supply the `Force` parameter.
+
+```
```
Comment (twitchax) [inline documentation/development-docs/should-process-confirm-impact.md]: Why not add `powershell` to the code fence?

File: README.md [hunk 3]
```diff
@@ -23,219 +20,148 @@ Below is a table containing the various Azure PowerShell rollup modules found in
...
 
-If you use both mechanisms on the same subscription, Microsoft Azure Active Directory authentication always wins. If you want to go back to management certificate authentication, please use ```Remove-AzureAccount```, which will remove the Microsoft Azure Active Directory information and bring management certificate authentication back in.
+### Log in to Azure
 
-#### Login directly from PowerShell (Microsoft Azure Active Directory authentication)
+To connect to Azure, use the [`Add-AzureRmAccount`](https://docs.microsoft.com/en-us/powershell/module/azurerm.profile/add-azurermaccount) cmdlet.
 
 ```powershell
-# Interactive login - you will get a dialog box asking for your Azure credentials
+# Interactive log in - you will get a diaglog box asking for your Azure credentials
```
Comment (maddieclayton) [inline README.md]: nit: dialog

File: documentation/development-docs/azure-powershell-design-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+### Closed Set of Values
+
+If there is a closed set of values applicable for a given parameter, use either a `ValidateSet` or an enumeration type. This functionality allows users to tab through the different values they can provide. From the [Strongly Encouraged Development Guidelines](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx):
+
+> _There are two ways to create a parameter whose value can be selected from a set of options._
+> 
+> - _Define an enumeration type (or use an existing type) that specifies the valid values. Then, use the enumeration type to create a parameter of that type._
+> - _Add the **ValidateSet** attribute to the parameter declaration._
+
```
Comment (maddieclayton) [inline documentation/development-docs/azure-powershell-design-guidelines.md]: Maybe explain Mandatory and Position? Unless you think that is self-explanatory.

File: documentation/development-docs/azure-powershell-design-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+To ensure that `-AsJob` is not broken in future changes, please add a test for this parameter. To update tests to include this parameter, use the following pattern:
+
+````powershell
+$job = Get-AzureRmSubscription
+$job | Wait-Job
+$subcriptions = $job | Receive-Job
+````
+
+## Argument Completers
```
Comment (maddieclayton) [inline documentation/development-docs/azure-powershell-design-guidelines.md]: This probably goes best right under ValidateSet

File: documentation/development-docs/azure-powershell-design-guidelines.md [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+
+### Required Parameter Sets
+
+In most Azure PowerShell cmdlets, there is a bare minimum of three parameter sets that need to be implemented.
+
+#### Interactive Parameter Set
+
+This parameter set should be implemented by _every_ cmdlet - in most cases, the user provides the name of the resource that they are acting upon (`Name`) and the resource group in which they are acting in (`ResourceGroupName`).
+
+The interactive parameter set **will always be the default parameter set** for a cmdlet. This means that when PowerShell is unable to determine which parameter set a user is in, it will default to the interactive parameter set and prompt the user to provide values for the missing mandatory parameters.
```
Comment (maddieclayton) [inline documentation/development-docs/azure-powershell-design-guidelines.md]: Probably should state that they need to explicitly set the interactive parameter set as the DefaultParameterSet.

File: documentation/development-docs/azure-powershell-developer-guide.md [hunk 7]
```diff
@@ -0,0 +1,451 @@
...
+
+# Before Adding a New Project
+
+## .NET SDK
+
+Before adding a new project to Azure PowerShell, you must have generated an [SDK for .NET](https://github.com/Azure/azure-sdk-for-net) using [AutoRest](https://github.com/Azure/autorest) for your service, and it must have been merged into the repository. 
+
+For more information about on-boarding a new library in the SDK for .NET repository, click [here](https://github.com/Azure/azure-sdk-for-net#to-on-board-new-libraries).
+
+## Design Review
```
Comment (maddieclayton) [inline documentation/development-docs/azure-powershell-developer-guide.md]: Update this section with the new guidelines.

File: documentation/development-docs/azure-powershell-developer-guide.md [hunk 8]
```diff
@@ -0,0 +1,451 @@
...
+- `Commands.Common.Network`
+    - Found in `src/ResourceManager/Common/Commands.Common.Network`
+- `Commands.Common.Storage`
+    - Found in `src/ResourceManager/Common/Commands.Common.Storage`
+
+# Creating Cmdlets
+
+## PowerShell Cmdlet Design Guidelines
+
+Please check out the [PowerShell Cmdlet Design Guidelines](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines) page for more information on how to create cmdlets that follow the PowerShell guidelines.
```
Comment (maddieclayton) [inline documentation/development-docs/azure-powershell-developer-guide.md]: Update this link to your new doc

File: documentation/development-docs/azure-powershell-developer-guide.md [hunk 9]
```diff
@@ -0,0 +1,451 @@
...
+
+To ensure that `-AsJob` is not broken in future changes, please add a test for this parameter. To update tests to include this parameter, use the following pattern:
+
+````powershell
+$job = Get-AzureRmSubscription
+$job | Wait-Job
+$subcriptions = $job | Receive-Job
+````
+
+## Argument Completers
```
Comment (maddieclayton) [inline documentation/development-docs/azure-powershell-developer-guide.md]: This feels unnecessary here

File: documentation/testing-docs/repo-tasks-module.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+## Usage:
+
+1. Double-click the shortcut located at `.\tools\PS-VSPrompt.lnk`, which will start VS Developer Prompt in PowerShell
+2. Run the command `Import-Module .\Repo-Tasks.psd1` to import the `Repo-Tasks` module
+	- During import, we allow users to load additional functions that they might want to use it in their session
+	- If you have a `userPreference.ps1` file under the `%userprofile%/psFiles` directory, the `Repo-Tasks` module will try to load it by dot sourcing it
+	- The module will also honor the environment variable `$env:psuserpreferences` and load `.ps1` files from the location that is pointed by `$env:psuserpreferences`
+	- As long as you have exported all functions that you need from your `.ps1` file using `Export-ModuleMember -Function {{functionName}}`. We deliberately do this to avoid polluting the list of commands available (when you use `Get-Command`)
+3. Currently, the `Repo-Tasks` module supports the following tasks:
+	- `Set-TestEnvironment`
```
Comment (maddieclayton) [inline documentation/testing-docs/repo-tasks-module.md]: Set-TestEnvironment and New-TestCredential

File: documentation/development-docs/should-process-confirm-impact.md (hunk 1 above)
Comment (cormacpayne) [reply documentation/development-docs/should-process-confirm-impact.md]: @twitchax `cs` is the name for C# blocks

File: documentation/development-docs/should-process-confirm-impact.md (hunk 2 above)
Comment (cormacpayne) [reply documentation/development-docs/should-process-confirm-impact.md]: @twitchax fixed

File: README.md (hunk 3 above)
Comment (cormacpayne) [reply README.md]: @maddieclayton fixed

File: documentation/development-docs/azure-powershell-design-guidelines.md (hunk 4 above)
Comment (cormacpayne) [reply documentation/development-docs/azure-powershell-design-guidelines.md]: @maddieclayton I think these properties are self-explanatory, but I will create a follow-up document that explains the attributes and properties found in PowerShell from `System.Management.Automation`, similar to the examples we have [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines)

File: documentation/development-docs/azure-powershell-design-guidelines.md (hunk 6 above)
Comment (cormacpayne) [reply documentation/development-docs/azure-powershell-design-guidelines.md]: @maddieclayton fixed

File: documentation/development-docs/azure-powershell-design-guidelines.md (hunk 5 above)
Comment (cormacpayne) [reply documentation/development-docs/azure-powershell-design-guidelines.md]: @maddieclayton leaving this section at the bottom and linking out to _Argument Completers_  section from the _Closed Set of Values_ section

File: documentation/development-docs/azure-powershell-developer-guide.md (hunk 7 above)
Comment (cormacpayne) [reply documentation/development-docs/azure-powershell-developer-guide.md]: @maddieclayton fixed

File: documentation/development-docs/azure-powershell-developer-guide.md (hunk 8 above)
Comment (cormacpayne) [reply documentation/development-docs/azure-powershell-developer-guide.md]: @maddieclayton fixed

File: documentation/development-docs/azure-powershell-developer-guide.md (hunk 9 above)
Comment (cormacpayne) [reply documentation/development-docs/azure-powershell-developer-guide.md]: @maddieclayton I plan on moving this information to the new document outlining PowerShell attributes and properties

File: documentation/testing-docs/repo-tasks-module.md (hunk 10 above)
Comment (cormacpayne) [reply documentation/testing-docs/repo-tasks-module.md]: @maddieclayton fixed

**Comment by MiYanni** [comment]
- Some kind of Output Format Guide related to the .format.ps1 files. More specifically, defaulting to table format.
- In addition to Set vs. Update, an Add vs. New would be great too.

File: documentation/development-docs/should-process-confirm-impact.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+Users who specify the `-Confirm` parameter on the command prompt will automatically be prompted for **any** call to `ShouldProcess`, and can decide whether to continue processing. Users can increase or decrease their `$ConfirmPreference` level to decide which kinds of changes will be confirmed automaticlaly without user interaction.
+
+Users can specify the `-WhatIf` parameter to see all the `ShouldProcess` prompts that would occur during exeuction of a cmdlet, without actually making any changes.
+
+Some cmdlets required additional confirmation. For example, if a cmdlet would destroy existing resources in some circumstances, the cmdlet might detect that condition and prompt the user to verify before continuing. Overwriting an existing resource during resource creation, overwriting a file when downloading data, deleting a resource that is currently in use, or deleting a container that contains additional resources are all example of this pattern. To implement additional confirmation, and allow scripts to opt out of additional prompts, the above pattern is enhanced with calles to `ShouldContinue()` and the `Force` parameter:
```
Comment (MiYanni) [inline documentation/development-docs/should-process-confirm-impact.md]: Typo:  enhanced with **calles** to `ShouldContinue()` and
Fixed: enhanced with **calls** to `ShouldContinue()` and

**Pull request merged**